net/netns/netns_darwin.go

```diff
@@ -34,15 +42,109 @@ func controlLogf(logf logger.Logf, network, address string, c syscall.RawConn) e
 		// Don't bind to an interface for localhost connections.
 		return nil
 	}
-	idx, err := interfaces.DefaultRouteInterfaceIndex()
+
+	addr, err := netip.ParseAddr(address)
 	if err != nil {
-		logf("[unexpected] netns: DefaultRouteInterfaceIndex: %v", err)
+		logf("[unexpected] netns: error parsing address %q: %v", address, err)
 		return nil
 	}
 
+	var idx int
+	if bindToDeviceByRoute() {
+		idx, err = interfaceIndexFor(addr)
+
+		// Verify that we didn't just choose the Tailscale interface;
+		// if so, we fall back to binding from the default.
+		if err == nil {
```

Comment:
Add logging if we get an error from `interfaceIndexFor`?

Reply:
Added 👍